### PR TITLE
Do not retry for certain types of errors

### PR DESCRIPTION
**Pull request by mnmtanish**

### CODE REVIEW

File: lib/kadira.js [hunk 1]
```diff
@@ -212,23 +212,49 @@ Kadira._send = function (payload, callback) {
...
-          err = new Error("Status code: " + response.statusCode);
+          err = new Error("Kadira: Unknown server error: " + status);
         }
-        tryAgain(err);
+
+        return tryAgain(err);
+      }
+
+      // do not retry for kadira agent errors
+      if(status === 429 || status === 430) {
```
Comment (arunoda) [inline lib/kadira.js]: We don't need to tell which kind of error here. Just printing the statusCode is enough. 

The we can have another page with known errors. 401 is a special case.

**Comment by mnmtanish** [comment]
Just the status code

File: lib/kadira.js [hunk 2]
```diff
@@ -212,23 +212,47 @@ Kadira._send = function (payload, callback) {
...
+          err = new Error("Kadira: Unknown server error: " + status);
         }
-        tryAgain(err);
+
+        return tryAgain(err);
+      }
+
+      // do not retry for kadira agent errors
+      if(status >= 400 && status < 500) {
+        console.error('Kadira: Unknown kadira agent error:', status);
```
Comment (arunoda) [inline lib/kadira.js]: Message needs to be just agent error. Not unknown agent error. 

Saying unknown is not good I think.

Comment (mnmtanish) [inline lib/kadira.js]: Updated

**Comment by arunoda** [comment]
LGTM

**Pull request merged**